core/orchestrator.go

```diff
@@ -453,12 +453,22 @@ type RemoteTranscoderFatalError struct {
 }
 
 var RemoteTranscoderTimeout = 8 * time.Second
+var ErrRemoteTranscoderTimeout = errors.New("Remote transcoder took too long")
 
 func (r *RemoteTranscoder) Capacity() int {
 	return r.capacity
 }
 
 func (rt *RemoteTranscoder) Transcode(fname string, profiles []ffmpeg.VideoProfile) ([][]byte, error) {
+	signalEof := func(err error) ([][]byte, error) {
+		// select so we don't block indefinitely if there's no listener
+		select {
```

Comment:
I'm confused on the syntax here. We're not executing anything in the case of `rt.eof <- struct{}{}:` or in the case of `default:`. Why is it important to have that statement there in the first place?

Reply:
Yeah, this is a golang-ism that stems from the behavior of go channels.

Basically, writing to a channel blocks if the channel is full (or unbuffered without a corresponding listener, as we discovered in https://github.com/livepeer/go-livepeer/pull/852). We can avoid that blocking by putting the channel write (`rt.eof <- struct{}{}`) into a `select` , which will fall through to the default case if the write will block.

Encountered this block while unit-testing. Basically we can't assume the `eof` channel will always have an external listener attached to it. Eg, with two concurrent transcodes, if the first one fails, the second will hang since the first `eof` has been consumed.

Reply:
I see what you're saying. If we have a transcoder with capacity of 5, for example, and have two transcodes happening at once, the second won't "end". Would another solution be to just make the eof channel a length of the transcoder's capacity? Or is this solution better since we don't have to specify that?

Reply:
Yeah, that approach of increasing the buffer size would probably work (and is what was used in https://github.com/livepeer/go-livepeer/pull/852)- note that `eof` itself is buffered with size 1.

Since a buffered / bounded channel can still block if the buffer was full for any reason, I think I still prefer this approach because there's no possibility of blocking.

Reply:
Yea that makes sense to me. But, I do have to say I'm proud of myself for thinking of an alternate solution relevant to some of the fixes we've seen in the last couple of days 😄